### PR TITLE
232 result form file upload validation

### LINKED ISSUE #232: Add more error validation when creating a result form through uploading a result form file in a tally
Add checks for the following:

- [ ] Check that the assigned center, ballot and station are active.

- [ ] Check that the assigned station exist.

- [ ] Check ballot number and center sub constituency code match when center sub constituency exist.

This issue is related to #229

### PR DESCRIPTION
**Pull request by JohnMwashuma**
Fixes #232

### CODE REVIEW

File: tally_ho/apps/tally/management/commands/import_data.py [hunk 1]
```diff
@@ -318,19 +318,73 @@ def process_results_form_row(tally, row, command=None, logger=None):
     ballot_number, code, station_number, gender, name,\
         office_name, _, barcode, serial_number = row[0:9]
 
-    ballot = Ballot.objects.get(number=ballot_number, tally=tally)
     gender = gender and getattr(Gender, gender.upper())
+    ballot = None
+
+    try:
+        ballot = Ballot.objects.get(number=ballot_number, tally=tally)
+        if ballot and not ballot.active:
```
Comment (pld) [inline tally_ho/apps/tally/management/commands/import_data.py]: I'm not sure it's possible to be in a scenario where `ballot` is null, wouldn't that have lead to a `DoesNotExist` exception?

File: tally_ho/apps/tally/management/commands/import_data.py [hunk 2]
```diff
@@ -318,19 +318,73 @@ def process_results_form_row(tally, row, command=None, logger=None):
...
+        if command:
+            command.stdout.write(command.style.WARNING(msg))
+        if logger:
+            logger.warning(msg)
+
     center = None
 
     try:
         center = Center.objects.get(code=code, tally=tally)
+        if center and not center.active:
```
Comment (pld) [inline tally_ho/apps/tally/management/commands/import_data.py]: same as above

File: tally_ho/apps/tally/management/commands/import_data.py [hunk 3]
```diff
@@ -318,19 +318,73 @@ def process_results_form_row(tally, row, command=None, logger=None):
...
         if command:
             command.stdout.write(command.style.WARNING(msg))
         if logger:
             logger.warning(msg)
 
+    try:
+        station = Station.objects.get(
+            station_number=station_number,
+            center=center)
+        if station and not station.active:
```
Comment (pld) [inline tally_ho/apps/tally/management/commands/import_data.py]: same as above

File: tally_ho/apps/tally/management/commands/import_data.py (hunk 1 above)
Comment (JohnMwashuma) [reply tally_ho/apps/tally/management/commands/import_data.py]: Nice catch, will change this

**Pull request merged**